railties/test/generators/actions_test.rb

```diff
@@ -494,57 +494,59 @@ def test_env_option_should_win_over_rails_env_variable_when_running_rails
     end
   end
 
-  def test_route_should_add_data_to_the_routes_block_in_config_routes
+  test "route should add route" do
     run_generator
-    route_command = "route '/login', controller: 'sessions', action: 'new'"
-    action :route, route_command
-    assert_file "config/routes.rb", /#{Regexp.escape(route_command)}/
+    route_commands = ["get 'foo'", "get 'bar'", "get 'baz'"]
+    route_commands.each do |route_command|
+      action :route, route_command
+    end
+    assert_routes route_commands
   end
 
-  def test_route_should_be_idempotent
+  test "route should indent routing code" do
     run_generator
-    route_path = File.expand_path("config/routes.rb", destination_root)
-
-    # runs first time, not asserting
-    action :route, "root 'welcome#index'"
-    content_1 = File.read(route_path)
-
-    # runs second time
-    action :route, "root 'welcome#index'"
-    content_2 = File.read(route_path)
-
-    assert_equal content_1, content_2
+    route_commands = ["get 'foo'", "get 'bar'", "get 'baz'"]
+    action :route, route_commands.join("\n")
+    assert_routes route_commands
   end
 
-  def test_route_should_add_data_with_an_new_line
+  test "route should be idempotent" do
     run_generator
-    action :route, "root 'welcome#index'"
-    route_path = File.expand_path("config/routes.rb", destination_root)
-    content = File.read(route_path)
+    route_command = "root 'welcome#index'"
 
-    # Remove all of the comments and blank lines from the routes file
-    content.gsub!(/^  \#.*\n/, "")
-    content.gsub!(/^\n/, "")
-
-    File.write(route_path, content)
+    # runs first time
+    action :route, route_command
+    assert_routes route_command
 
-    routes = <<-F
-Rails.application.routes.draw do
-  root 'welcome#index'
-end
-F
+    content = File.read(File.expand_path("config/routes.rb", destination_root))
 
-    assert_file "config/routes.rb", routes
+    # runs second time
+    action :route, route_command
+    assert_file "config/routes.rb", content
+  end
 
-    action :route, "resources :product_lines"
+  test "route with namespace option should nest route" do
+    run_generator
+    action :route, "get 'foo'\nget 'bar'", namespace: :baz
```

Comment:
This will cause keyword arguments waring in Ruby 2.7. 

```ruby 
$ ruby -v
ruby 2.7.0dev (2019-11-15T09:07:34Z master 11ae47c266) [x86_64-linux]
$ ./bin/test -w test/generators/actions_test.rb:541
Run options: --seed 43932

# Running:

/home/y-yagi/src/github.com/rails/rails/railties/test/generators/actions_test.rb:587: warning: The last argument is used as the keyword parameter
/home/y-yagi/src/github.com/rails/rails/railties/lib/rails/generators/actions.rb:258: warning: for `route' defined here
.

```

Reply:
Hmm...  It looks like the way we've been handling this (e.g. in 048436213e923e5c4811059fd660c9042f6f19d1 and de3d9680ac5a0667d3ba885cd3d2a394639fe538) is a conditional on `RUBY_VERSION`.  So something like:
```ruby
if RUBY_VERSION < "2.7"
  def action(*args, &block)
    capture(:stdout) { generator.send(*args, &block) }
  end
else
  def action(*args, **kwargs, &block)
    capture(:stdout) { generator.send(*args, **kwargs, &block) }
  end
end
```

Which is a bit ugly...

Alternatively, I could make the `route` accept an `options` Hash instead of kwargs, but that doesn't seem right either.

Reply:
How about this?  

```diff 
-    def action(*args, &block)
-      capture(:stdout) { generator.send(*args, &block) }
+    def action(*args, **kwargs, &block)
+      if kwargs.empty?
+        capture(:stdout) { generator.send(*args, &block) }
+      else
+        capture(:stdout) { generator.send(*args, **kwargs, &block) }
+      end
     end
```

This is still a bit ugly. But I think this is better than to check Ruby version.

Reply:
Yes, that looks less ugly.  Though the drawback is the code may not be found when we drop support for Ruby 2.6 in the future and grep for code to remove.  (Or if Ruby 2.7 kwargs handling changes before release, and we grep for code to revert.)

Reply:
Basically, we should not use a version to check a feature. 
There is a possibility that breaks in implementations other than CRuby(Also, there is a possibility that breaks in a new version of CRuby).
But unfortunately, other codes have the same issue. I will check the whole later.